code/go/0chain.net/smartcontract/storagesc/allocation_test.go

```diff
@@ -230,6 +232,219 @@ func TestSelectBlobbers(t *testing.T) {
+				ID:       ba.BlobberID,
+				Capacity: mockBlobberCapacity,
+				Terms: Terms{
+					MaxOfferDuration: mockMaxOffDuration,
+					ReadPrice:        mockReadPrice,
+					WritePrice:       mockWritePrice,
+				},
+				LastHealthCheck: now,
+			}
+			blobbers = append(blobbers, blobber)
+		}
+
+		alloc := &StorageAllocation{
+			ID:              mockAllocationID,
+			Owner:           mockOwner,
+			BlobberDetails:  blobberAllocation,
+			BlobberMap:      blobberMap,
+			Name:            mockAllocationName,
+			Size:            confMinAllocSize,
+			Expiration:      mockAllocationExpiry,
+			ReadPriceRange:  PriceRange{mockMinPrice, mockMaxPrice},
+			WritePriceRange: PriceRange{mockMinPrice, mockMaxPrice},
+			DataShards:      arg.dataShards,
+			ParityShards:    arg.parityShards,
+		}
+
+		if len(arg.addBlobberID) > 0 {
+			blobber := &StorageNode{
+				ID: arg.addBlobberID,
+				Terms: Terms{
+					MaxOfferDuration: mockMaxOffDuration,
+					ReadPrice:        mockReadPrice,
+					WritePrice:       mockWritePrice,
+				},
+				Capacity:        mockBlobberCapacity,
+				LastHealthCheck: now,
+			}
+			balances.On("GetTrieNode", blobber.GetKey(sc.ID), mock.MatchedBy(func(s *StorageNode) bool {
+				*s = *blobber
+				return true
+			})).Return(nil).Once()
+
+			sp := stakePool{
+				StakePool: stakepool.StakePool{
+					Pools: map[string]*stakepool.DelegatePool{
+						mockPoolId: {},
+					},
+				},
+			}
+			sp.Pools[mockPoolId].Balance = mockState
+			balances.On("GetTrieNode", stakePoolKey(sc.ID, arg.addBlobberID),
+				mock.MatchedBy(func(s *stakePool) bool {
+					*s = sp
+					return true
+				})).Return(nil).Twice()
+		}
+
+		if len(arg.removeBlobberID) > 0 {
+			blobber := &StorageNode{
+				ID:       arg.removeBlobberID,
+				Capacity: mockBlobberCapacity,
+				Terms: Terms{
+					MaxOfferDuration: mockMaxOffDuration,
+					ReadPrice:        mockReadPrice,
+					WritePrice:       mockWritePrice,
+				},
+				LastHealthCheck: now,
+			}
+
+			balances.On(
+				"InsertTrieNode",
+				blobber.GetKey(sc.ID),
+				mock.Anything,
+			).Return("", nil).Once()
+
+			balances.On(
+				"EmitEvent",
+				event.TypeStats,
+				event.TagUpdateBlobber,
+				blobber.ID,
+				mock.Anything,
+			).Return("", nil).Once()
+		}
+
+		return blobbers, arg.addBlobberID, arg.removeBlobberID, &sc, alloc, now, balances
+
+	}
+
+	testCases := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "test_blobber_remove_doesnt_exist",
+			args: args{
+				numBlobbers:     6,
+				addBlobberID:    "add_blobber_id",
+				removeBlobberID: "blobber_non_existent",
+				dataShards:      5,
+			},
+			want: want{
+				err:    true,
+				errMsg: "cannot find blobber blobber_non_existent in allocation",
+			}},
+		{
+			name: "test_blobber_add_remove_blobber",
+			args: args{
+				numBlobbers:     6,
+				addBlobberID:    "add_blobber_id",
+				removeBlobberID: "blobber_1",
+				dataShards:      5,
+			},
+			want: want{
+				err: false,
+			}},
```

Comment:
Please add more test cases, we should cover most of the process logic of the changeBlobbers method.